clash-cores/src/Clash/Cores/Xilinx/Xpm/Cdc/Gray.hs

```diff
@@ -0,0 +1,103 @@
+{-|
+  Copyright   :  (C) 2023, Google LLC
+  License     :  BSD2 (see the file LICENSE)
+  Maintainer  :  QBayLogic B.V. <devops@qbaylogic.com>
+-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE RecordWildCards #-}
+
+-- See [Note: eta port names for tdpbram]
+{-# OPTIONS_GHC -fno-do-lambda-eta-expansion #-}
```

Comment:
Why do you do this in combination with the NOINLINE over at
https://github.com/clash-lang/clash-compiler/pull/2452/files#diff-e9f756a2ad405040ed110e5eca448f27ac331270f762b039437671580e660149R103?

Why not just tag `xpmCdcGrayWith` with an `INLINE` instead?

The reason I'm asking is the reason that we do https://github.com/clash-lang/clash-compiler/blob/95ee5b9421991abfea0783e9295e09c2383ad87f/clash-prelude/src/Clash/Explicit/BlockRam.hs#L1217 is because of https://github.com/clash-lang/clash-compiler/blob/95ee5b9421991abfea0783e9295e09c2383ad87f/clash-prelude/src/Clash/Explicit/BlockRam.hs#L1223

Is that the case for this XPM_CDC_GRAY as well? That Vivado complains when XPM_CDC_GRAY does not live in its own architecture?

Reply:
If that is the case, I feel you should also link `[Note: true dual-port blockRAM separate architecture]` because the `NOINLINE` on `xpmCdcGrayWith` is a bit surprising and should be documented.

Reply:
This is _just_ for pretty names in the HDL. I guess it's not that relevant given that `XPM_CDC_GRAY` has pretty names to begin with. Vivado does not complain about anything if we remove the flag + NOINLINE. I assumed the NOINLINE was necessary for Clash to pick up on the arg / port names.

Does this answer your question? Would you like it removed instead given the macro already has pretty port names?

Reply:
Yes, I’d like the ghc option to be removed and have `xpmCdcGrayWith` marked `INLINE` instead.

Reply:
It should probably also be eta-reduced then.

Reply:
Done